sewer/auth.py

```diff
@@ -1,42 +1,115 @@
 import logging
+from typing import Sequence, Tuple
 
 
-class BaseAuthProvider(object):
-    def __init__(self, auth_type, LOG_LEVEL="INFO"):
-        self.LOG_LEVEL = LOG_LEVEL
-        self.dns_provider_name = self.__class__.__name__
+class ValidationMethod:
+    def __init__(self, label: str, id_type: str, supported: bool):
+        self.label = label
+        self.id_type = id_type
 
-        self.logger = logging.getLogger(__name__)
-        self.logger.setLevel(self.LOG_LEVEL)
-        if not self.logger.hasHandlers():
-            handler = logging.StreamHandler()
-            formatter = logging.Formatter("%(message)s")
-            handler.setFormatter(formatter)
-            self.logger.addHandler(handler)
 
-        self.auth_type = auth_type
+supported_validations = {
+    "http-01": ValidationMethod("http-01", "dns", True),
```

Comment:
should `"dns", True` for this?

Reply:
tl;dr: ACME has two type: fields for different things.  Stop now unless you enjoy the pain of perusing RFCs over and over.  :-)

Yes.  Though the True at the end is either an artifact of history not recorded in a commit, or an accommodation for a future change (list of all known, whether supported or not?) not yet implemented.

The "dns" is the authorization type, and is short for "domain-name", the sort sewer knows about to date.  Recently an "ip" type has been added.  I'm not certain of the use for that except for an impression that it's aimed at hosting providers and other large-scale services.

The familiar "http-01" and "dns-01" (and that new tls one whose spelling I don't care to lookup now) identify the challenge type, which might better be called the challenge protocol.  The RFCs - more than just 8555 is involved, and there's a registry - add to the confusion by calling the challenge type "label", as in rfc8555 9.7.7:

`Label: The value to be put in the "type" field of the identifier object`

There's an example of the result of querying an unsatisfied authorization somewhere in the RFC - that's why we have to sift through them to find the one that is for dns-01 (or now, http-01) to get the magic values used in forming the challenge response.